tests/entrypoints/openai/test_tensorizer_entrypoint.py

```diff
@@ -0,0 +1,114 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import tempfile
+import weakref
+
+import openai
+import pytest
+import pytest_asyncio
+import torch.cuda
+
+from vllm import LLM, SamplingParams
+from vllm.engine.arg_utils import EngineArgs
+from vllm.lora.request import LoRARequest
+from vllm.model_executor.model_loader.tensorizer import (
+    TensorizerConfig, tensorize_lora_adapter, tensorize_vllm_model)
+
+from ...utils import RemoteOpenAIServer
+
+MODEL_NAME = "meta-llama/Llama-2-7b-hf"
+LORA_PATH = "yard1/llama-2-7b-sql-lora-test"
+
+
+@pytest.fixture(scope="module")
+def tmp_dir():
+    tmp_dir = tempfile.TemporaryDirectory()
+
+    def cleanup():
+        tmp_dir.cleanup()
+
+    weakref.finalize(tmp_dir, cleanup)
+
+    yield tmp_dir
+
+    cleanup()
```

Comment:
A better pattern for this is usually to use `try` + `finally`, but in this case, `tempfile.TemporaryDirectory` is a context manager, so you can achieve the same effect with a `with` statement:
```suggestion
def tmp_dir():
    with tempfile.TemporaryDirectory() as path:
        yield path
```
(Note: this yields the equivalent of just `tmp_dir.name` as a `str` instead of the full stateful object)

Reply:
I guess `try + finally` is pretty functionally equivalent to a context manager due to having a guaranteed cleanup?